src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1211,12 +1211,13 @@ private void Login(ServerInfo server, TimeoutTimer timeout, string newPassword,
                 _sessionRecoveryRequested = true;
             }
 
-            // If the workflow being used is Active Directory Password or Active Directory Integrated and server's prelogin response
+            // If the workflow being used is Active Directory Password/Integrated/Interactive/Service Principal and server's prelogin response
             // for FEDAUTHREQUIRED option indicates Federated Authentication is required, we have to insert FedAuth Feature Extension
-            // in Login7, indicating the intent to use Active Directory Authentication Library for SQL Server.
-            if (ConnectionOptions.Authentication == SqlAuthenticationMethod.ActiveDirectoryPassword
+            // in Login7, indicating the intent to use Active Directory Authentication for SQL Server.
+            if ((ConnectionOptions.Authentication == SqlAuthenticationMethod.ActiveDirectoryPassword
```

Comment:
If you notice the parens () change here, it looks like a long-standing, minor bug in both netcore and netfx implementations. The previous logic did not make sense.